draft-ietf-quic-multipath.md

```diff
@@ -287,12 +346,15 @@ Furthermore, this document
 does not discuss when a client decides to initiate a new path. We
 delegate such discussion in separate documents.
 
+To let the peer open a new path, an endpoint needs to provide its peer with connection IDs
+and associated Path Identifiers for the new path. 
+
 To open a new path, an endpoint SHALL use different Connection IDs on different paths.
 Still, the receiver may observe the same Connection ID used on different
 4-tuples due to, e.g., NAT rebinding. In such case, the receiver reacts
 as specified in {{Section 9.3 of QUIC-TRANSPORT}}.
 
```

Comment:
We need to be a bit more precise there. RFC 9000 suggests that the server sends two path challenges, one to the old client address and one to the new one. We need to specify what connection ID it picks.

- for the old address, I assume the server should keep using the same MP_CONNECTION_ID, mapping to the same path ID.
- for the new address, the server SHOULD pick an MP_CONNECTION_ID that has not been used yet. We have to discuss whether that new MP_CONNECTION_ID should point to the same path or to a different path.

I would argue for "same path", for two reasons:

1. We don't want the server to initiate new paths. If it did, there could be a race condition with the client deciding to use that path ID for a different 4-tuple.
2. We don't want NAT rebinding to consume the path identifier budget at the server, because it is trivial for attackers to intercept a client's message, change the source IP, and forward the message to the server. If new path-ids are used each time, we have a path-id exhaustion attack.

Reply:
I agree, and this is more or less exactly how it is implemented in the Rask stack.

Reply:
I added a clarification above. Please have a look @huitema !

However, why do you think we need t send two path challenges? On mitigation you don't send a path challenge to the old address in RFC9000. If path validation fails you keep the old path (and retire the new connection ID). If migration succeeds, you can retire the old connection ID.